bin/firebase-bolt

```diff
@@ -15,24 +15,130 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
+'use strict';
 
+var fs = require('fs');
+var parseArgs = require('minimist');
+
+var util = require('../lib/util');
 var bolt = require('../lib/bolt');
+var pkg = require('../package.json');
+
+var VERSION_STRING = "Firebase Bolt v" + pkg.version;
+
+var opts = {
+  boolean: ['version', 'help'],
+  alias: {
+    'v': 'version',
+    'h': 'help',
+    'o': 'output',
+  },
+  unknown: function(flag) {
+    if (flag.slice(0, 2) == '--') {
+      log("Unknown flag: '" + flag + "'");
+      usage(1);
+    }
+  }
+};
+
+function main() {
+  var args = parseArgs(process.argv.slice(2), opts);
+
+  if (args.version) {
+    console.log(VERSION_STRING);
+    return;
+  }
+
+  if (args.help) {
+    usage(0);
+  }
+
+  if (args._.length > 1) {
+    log("Can only compile a single file.");
+    usage(1);
+  }
+
+  // Pipe Bolt file from stdin
+  if (args._.length === 0) {
+    readFile(process.stdin, function(data) {
+      console.log(translateRules(data));
+    });
+    return;
+  }
 
-var input = "";
+  // Read Bolt file and write json file.
+  var inFile = util.ensureExtension(args._[0], bolt.FILE_EXTENSION);
+  var outFile;
+  if (args.output) {
+    outFile = util.ensureExtension(args.output, 'json');
+  } else {
+    outFile = util.replaceExtension(inFile, 'json');
+  }
+  log("Generating " + outFile + ".");
+  if (inFile === outFile) {
+    log("Cannot overwrite input file: " + inFile);
+    log("(Did you mean '" + util.replaceExtension(inFile, 'bolt') + "'?)");
+    process.exit(1);
+  }
+  fs.readFile(inFile, 'utf8', function(err, data) {
```

Comment:
In a CLI like this, it's totally reasonable to use the synchronous file system commands. You've obviously already implemented async which is technically superior, just a random note and a shortcut I often take.